### PR TITLE
Upgrade tally and updates to match StatsReporter, Gauge, Closer changes

### PR DESCRIPTION
**Pull request by madhuravi**
NOTE: I have a TODO for the stats reporter. Will add that in a later PR

### CODE REVIEW

File: service/host.go [hunk 1]
```diff
@@ -156,7 +156,9 @@ func (s *host) shutdown(err error, reason string, exitCode *int) (bool, error) {
 
 	// Stop the metrics reporting
 	if s.scope != nil {
-		s.scope.Close()
+		if err = s.metricsCloser.Close(); err != nil {
```
Comment (sectioneight) [inline service/host.go]: small possibility for an upstream bug to affect us -- `s.metricsCloser` could be `nil` even if `s.scope` is non-nil. Perhaps we should be defensive and change line 158 to also check if `s.metricsCloser` is non-nil?

File: tracing/tracer.go [hunk 2]
```diff
@@ -41,8 +41,9 @@ func InitGlobalTracer(
 	var reporter *jaegerReporter
 	if cfg == nil || !cfg.Disabled {
 		cfg = loadAppConfig(cfg, logger)
+		// TODO: Change to use the right stats reporter
 		reporter = &jaegerReporter{
-			reporter: scope.Reporter(),
+			reporter: tally.NullStatsReporter,
```
Comment (sectioneight) [inline tracing/tracer.go]: oh, is this not possible anymore? are you going to address that in this PR or a future one?

File: service/host.go (hunk 1 above)
Comment (madhuravi) [inline service/host.go]: good catch.

File: tracing/tracer.go (hunk 2 above)
Comment (madhuravi) [inline tracing/tracer.go]: see description, to be addressed in future PR.

Comment (madhuravi) [inline tracing/tracer.go]: sorry I tend to interchange the two. Fixed description.

**Pull request merged**